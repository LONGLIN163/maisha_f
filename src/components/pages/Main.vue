<template>
    <div>
        <div class="main-div">
         <!-- use "router-view" to put the sub pages to the current page -->
            <keep-alive>
                <router-view />
            </keep-alive>
        </div>

        <van-tabbar v-model="active" @change="changeTabbar(active)">
            <van-tabbar-item icon="shop">Home</van-tabbar-item>
            <van-tabbar-item icon="records" data-test="tab-list">List</van-tabbar-item>
            <van-tabbar-item icon="cart">Shopping Cart</van-tabbar-item>
            <van-tabbar-item icon="contact" info="3" data-test="tab-account">Account</van-tabbar-item>
        </van-tabbar>

    </div>
</template>

<script>
    export default {
        data() {
            return {
                active: 0
            }
        },
        created(){
            this.changeTabBarActive()
        },
        updated(){
            this.changeTabBarActive()
        },
        methods: {
            changeTabbar(active) {
                console.log(active)
                switch(active){
                    case 0:
                        this.$router.push({name:'ShoppingMall'})
                        break;
                    case 1:
                        this.$router.push({name:'CategoryList'})
                        break;
                    case 2:
                        this.$router.push({name:'Cart'})
                        break;
                    case 3:
                        this.$router.push({name:'Member'})
                        break;
                    case 4:
                        break;
                }
            },
            changeTabBarActive(){
                this.nowPath=this.$route.path
                if(this.nowPath=='/cart'){
                    this.active=2;
                }
            },
        },
    }
</script>

<style lang="scss" scoped>

    .van-tabbar-item--active{
        color: $color-primary !important;
    }
    .van-info{
        color: $color-primary !important;
    }


</style>