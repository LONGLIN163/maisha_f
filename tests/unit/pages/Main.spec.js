
import { mount, createLocalVue} from '@vue/test-utils'
import Main from '@/components/pages/Main.vue'

import VueRouter from 'vue-router';
const Vant=require("vant")

const localVue = createLocalVue();
localVue.use(VueRouter); 
localVue.use(Vant); 

const router = new VueRouter();

describe('Main',  () => {
  
  let wrapper;
  //mock changeTabBarActive() and hook it to the component
  const changeTabBarActive = jest.spyOn(Main.methods, 'changeTabBarActive')
    .mockImplementation(() => {
      console.log("1. mock changeTabBarActive implementation")
    })
  const changeTabbar = jest.spyOn(Main.methods, 'changeTabbar')
    .mockImplementation(() => {
      console.log("2. mock changeTabbar implementation")
    })

  beforeEach(() => {
    wrapper = mount(Main,{
        localVue,
        router
      })
    })

    it('renders Main snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('changeTabBarActive() should has been called when running create()', () => {
       // changeTabBarActive will be called immediately after vm created.
      expect(changeTabBarActive).toHaveBeenCalled()
    })

    it('changeTabbar() should have been called correctly ',async () => {
      // click list tab
      await wrapper.find('[data-test="tab-list"]').trigger('click');
      expect(changeTabbar).toHaveBeenCalled()
      // changeTabBarActive will be called in the updated() hooks method after click and dom updated.
      expect(changeTabBarActive).toHaveBeenCalled()
      // click home tab
      await wrapper.find('[data-test="tab-shop"]').trigger('click');
      expect(changeTabbar).toHaveBeenCalled()
      expect(changeTabBarActive).toHaveBeenCalled()

    })

})