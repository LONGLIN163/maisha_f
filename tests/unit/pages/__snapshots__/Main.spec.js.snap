// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders Main snapshot 1`] = `
<div>
  <div class="main-div">
    <!---->
  </div>
  <div class="van-hairline--top-bottom van-tabbar van-tabbar--fixed">
    <div class="van-tabbar-item van-tabbar-item--active" data-test="tab-shop">
      <div class="van-tabbar-item__icon"><i class="van-icon van-icon-shop">
          <!---->
        </i>
        <!---->
      </div>
      <div class="van-tabbar-item__text">Home</div>
    </div>
    <div class="van-tabbar-item" data-test="tab-list">
      <div class="van-tabbar-item__icon"><i class="van-icon van-icon-records">
          <!---->
        </i>
        <!---->
      </div>
      <div class="van-tabbar-item__text">List</div>
    </div>
    <div class="van-tabbar-item" data-test="tab-cart">
      <div class="van-tabbar-item__icon"><i class="van-icon van-icon-cart">
          <!---->
        </i>
        <!---->
      </div>
      <div class="van-tabbar-item__text">Shopping Cart</div>
    </div>
    <div class="van-tabbar-item" data-test="tab-account">
      <div class="van-tabbar-item__icon"><i class="van-icon van-icon-contact">
          <!---->
        </i>
        <div class="van-info">3</div>
      </div>
      <div class="van-tabbar-item__text">Account</div>
    </div>
  </div>
</div>
`;
